### ISSUE TITLE
Improve layout of "subject specific guides" block in Timeline

### ISSUE THREAD

**Issue by ragesoss** [WikiEducationFoundation/WikiEduDashboard#5707]
When using the assignment wizard for a 'research and write' assignment, instructors select which subjects they will be working on, and the timeline includes a block of corresponding links to PDFs about subject-specific concerns for each of those selected subjects:

![guides links](https://github.com/WikiEducationFoundation/WikiEduDashboard/assets/848483/d24e4e8b-425e-4629-bf18-f1baaff2b445)

The visual layout of these links is not very elegant. They should probably float and only go to multiple lines if necessary, and they should be spaced nicely from each other.

(A PR to fix this should include a variety of 'after' screenshots with varying numbers of links.)

**Comment by psankhe28** [comment]
It needs to remain on the same line and move to the next line only if the space on the current line is filled.
Is this the behavior that you are looking for?

I would like to work on this.

**Comment by ragesoss** [comment]
@psankhe28 please work on one issue at a time.

**Comment by ragesoss** [comment]
> It needs to remain on the same line and move to the next line only if the space on the current line is filled. Is this the behavior that you are looking for?
> 

Yes, that sounds right.

**Comment by psankhe28** [comment]
> @psankhe28 please work on one issue at a time.

Yes. I'll focus on one issue at a time from now on. Thank you for the guidance

**Comment by ragesoss** [comment]
@psankhe28 yes, that looks good. Can it be done in a way that does work for already-created courses?

**Comment by PR4NJ41** [comment]
@ragesoss I think you are referring to me. I will try but the problem is that the way it is implemented. When a course is created these blocks are created and it doesn't include a classname. If i make changes to the classname which are already present then it creates problem in other blocks since other are using the same class.

**Comment by ragesoss** [comment]
@PR4NJ41 yes, sorry, I meant to tag you.

Is this because the styling is being applied parent component (currently a `<p>`) rather than the individual `a.handout-link` instances? If there's no good way to do it based on just the .handout-link class, then I'm okay with it only applying to new courses.

**Comment by PR4NJ41** [comment]
@ragesoss I am able to apply this based on querySelector. If it is ok should i make a PR ?

**Comment by ragesoss** [comment]
Sure, I'd like to see both methods.

**Comment by PR4NJ41** [comment]
@ragesoss I have made another PR please review it. Thank you!

**Comment by AarohiSharma5** [comment]
Hi @ragesoss, can I work on this issue?

**Comment by AarohiSharma5** [comment]
I went through the issue, and realised how about if we remove the "<p>" tags in the wizard_timeline_manager.rb? 
Like here, if we use-------->

"def link_to_handout(logic_key)
    link_text = HANDOUTS[logic_key][0]
    url = HANDOUTS[logic_key][1]
    <<~LINK
      <a class="handout-link" href="#{url}" target="_blank">#{link_text}</a>
    LINK
  end"

instead of-------->

"def link_to_handout(logic_key)
    link_text = HANDOUTS[logic_key][0]
    url = HANDOUTS[logic_key][1]
    <<~LINK
      <p>
      <a class="handout-link" href="#{url}" target="_blank">#{link_text}</a>
      </p>
    LINK
  end" ?

as these tags forces inline stacking here? 
and then doing adjustments in the css, for proper spacing and alignments? 

before->

![Image](https://github.com/user-attachments/assets/dcdf853a-5516-4ccf-8a71-1195033256c6)

after-> 

![Image](https://github.com/user-attachments/assets/3903afd4-c81c-4403-bd2a-b4f3a1236495)
